File: ios/Classes/SigmobadPlugin.m
#import "SigmobadPlugin.h"
#import <WindSDK/WindSDK.h>
#import "SigmobLogUtil.h"
#import "SigmobAdEvent.h"
#import "SigmobRewardAd.h"
#import "SigmobAdNativeView.h"
#import "SigmobInterstitialAd.h"
#import "SigmobAdSplashView.h"

@implementation SigmobadPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"sigmobad"
                                     binaryMessenger:[registrar messenger]];
    SigmobadPlugin* instance = [[SigmobadPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
    //注册event
    [[SigmobAdEvent sharedInstance]  initEvent:registrar];
    //注册native
    [registrar registerViewFactory:[[SigmobAdNativeViewFactory alloc] initWithMessenger:registrar.messenger] withId:@"com.gstory.sigmobad/NativeView"];
    //注册splash
    [registrar registerViewFactory:[[SigmobAdSplashViewFactory alloc] initWithMessenger:registrar.messenger] withId:@"com.gstory.sigmobad/SplashView"];

}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    //初始化
    if ([@"register" isEqualToString:call.method]) {
        NSString *appId = call.arguments[@"iosId"];
        NSString *appKey = call.arguments[@"iosAppKey"];
        BOOL debug = [call.arguments[@"debug"] boolValue];
        BOOL personalized =  [call.arguments[@"personalized"] boolValue];
        [[SigmobLogUtil sharedInstance] debug:debug];
        [WindAds setDebugEnable:debug];
        WindAdOptions *option = [[WindAdOptions alloc] initWithAppId:appId appKey:appKey];
        //个性化推荐
        [WindAds setPersonalizedAdvertising:personalized ? WindPersonalizedAdvertisingOn : WindPersonalizedAdvertisingOff];
        [WindAds startWithOptions:option];
        //渠道id
        result(@YES);
        //sdk版本
    }else if([@"getSDKVersion" isEqualToString:call.method]){
        NSString *version = [WindAds sdkVersion];
        result(version);
        //预加载激励广告
    }else if([@"loadRewardAd" isEqualToString:call.method]){
        [[SigmobRewardAd sharedInstance] loadAd:call.arguments];
        result(@YES);
        //显示激励广告
    }else if([@"showRewardAd" isEqualToString:call.method]){
        [[SigmobRewardAd sharedInstance] showAd];
        result(@YES);
        //预加载插屏广告
    }else if([@"loadInterstitialAd" isEqualToString:call.method]){
        [[SigmobInterstitialAd sharedInstance] loadAd:call.arguments];
        result(@YES);
        //显示插屏广告
    }else if([@"showInterstitialAd" isEqualToString:call.method]){
        [[SigmobInterstitialAd sharedInstance] showAd];
        result(@YES);
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end

File: android/src/main/kotlin/com/gstory/sigmobad/SigmobadPlugin.kt
package com.gstory.sigmobad

import android.app.Activity
import android.content.Context
import androidx.annotation.NonNull
import com.gstory.sigmobad.interstitial.SigmobAdInterstitial
import com.gstory.sigmobad.native.SigmobAdNativeViewFactory
import com.gstory.sigmobad.reward.SigmobAdReward
import com.gstory.sigmobad.splash.SigmobAdSplashViewFactory
import com.gstory.sigmobad.utils.SigmobLogUtil
import com.sigmob.windad.WindAdOptions
import com.sigmob.windad.WindAds
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result


/** SigmobadPlugin */
class SigmobadPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
    private lateinit var channel: MethodChannel

    private var applicationContext: Context? = null
    private var mActivity: Activity? = null
    private var mFlutterPluginBinding: FlutterPlugin.FlutterPluginBinding? = null

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        mActivity = binding.activity
        //注册开屏广告
        mFlutterPluginBinding?.platformViewRegistry?.registerViewFactory(
            "com.gstory.sigmobad/SplashView",
            SigmobAdSplashViewFactory(mFlutterPluginBinding!!.binaryMessenger)
        )
        //注册信息流广告
        mFlutterPluginBinding?.platformViewRegistry?.registerViewFactory(
            "com.gstory.sigmobad/NativeView",
            SigmobAdNativeViewFactory(mFlutterPluginBinding!!.binaryMessenger)
        )
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        mActivity = binding.activity
    }

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel =
            MethodChannel(flutterPluginBinding.binaryMessenger, "sigmobad")
        channel.setMethodCallHandler(this)
        applicationContext = flutterPluginBinding.applicationContext
        mFlutterPluginBinding = flutterPluginBinding
        SigmobAdEventPlugin().onAttachedToEngine(flutterPluginBinding)
//        FlutterUnionadViewPlugin.registerWith(flutterPluginBinding,mActivity!!)
    }

    override fun onDetachedFromActivityForConfigChanges() {
        mActivity = null
    }

    override fun onDetachedFromActivity() {
        mActivity = null
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        //注册初始化
        when (call.method) {
            "register" -> {
                val appId = call.argument<String>("androidId")
                val appKey = call.argument<String>("androidAppKey")
                val debug = call.argument<Boolean>("debug")
                val personalized = call.argument<Boolean>("personalized")
                val ads = WindAds.sharedAds()
                   // 是否显示日志
                SigmobLogUtil.setAppName("sigmobad")
                SigmobLogUtil.setShow(debug!!)
                ads.isDebugEnable = debug
                //个性化推荐开关
                ads.isPersonalizedAdvertisingOn = personalized!!
                ads.startWithOptions(applicationContext, WindAdOptions(appId, appKey))
                result.success(ads.isInit)
            }
            //sdk版本
            "getSDKVersion" -> {
                result.success(WindAds.getVersion())
//                result.success("123")
            }
            //预加载激励广告
            "loadRewardAd" -> {
                SigmobAdReward.init(call.arguments as Map<String?, Any?>)
                result.success(true)
            }
            //展示激励广告
            "showRewardAd" -> {
                SigmobAdReward.showRewardAd()
                result.success(true)
            }
            //预加载插屏广告
            "loadInterstitialAd" -> {
                SigmobAdInterstitial.init(call.arguments as Map<String?, Any?>)
                result.success(true)
            }
            //展示插屏广告
            "showInterstitialAd" -> {
                SigmobAdInterstitial.showInterstitialAd()
                result.success(true)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
